src/visualizers/VisualizerHistogram.ts

```diff
@@ -0,0 +1,211 @@
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+import {VisualizerDefault} from './VisualizerDefault'
+import {natlog} from '../shared/math'
+
+class HistogramVisualizer extends VisualizerDefault {
+    name = 'Histogram'
+
+    binSize = 1
+    terms = 10
+    firstIndex = 1
+    linear = false
+
+    params = {
+        binSize: {
+            value: this.binSize,
+            forceType: 'integer',
+            displayName: 'Bin Size',
+            required: true,
+        },
+
+        firstIndex: {
+            value: this.firstIndex,
+            forceType: 'integer',
+            displayName: 'First Index',
+            required: true,
+        },
+
+        terms: {
+            value: this.terms,
+            forceType: 'integer',
+            displayName: 'How many terms of the series',
+            required: true,
+        },
+    }
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        if (this.params.binSize.value < 1) {
+            status.isValid = false
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        if (this.params.binSize.value < 1) {
+            status.isValid = false
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        if (this.params.firstIndex.value < 1) {
+            status.isValid = false
+            status.errors.push('First Index can not be less than 1')
+        }
+
+        return status
+    }
+
+    // Find the largest value of the sequence asked for.
+    largestValue(): bigint {
+        let largest_value = 0n
+        for (
+            let i = Math.max(this.firstIndex, this.seq.first);
+            i < Math.min(this.terms, this.seq.last);
+            i++
+        ) {
+            const value = this.seq.getElement(i)
+            if (i == 0) {
+                largest_value = value
```

Comment:
I don't think `i == 0` is the correct condition here, since if the user chose a firstIndex larger than zero, there never will be any such i. Also, as per the coding guide, prefer `===` over `==` when it is applicable, as it is here.